src/mds/CInode.cc

```diff
@@ -3732,21 +3728,17 @@ void CInode::validate_disk_state(CInode::validated_data *results,
       }
       if (in->is_symlink()) {
         // there's nothing to do for symlinks!
-        results->passed_validation = true;
         return true;
       }
 
-      results->passed_validation = false; // we haven't finished it yet
```

Comment:
Need to initialize this to false or if we bail out we'll have an undefined result.